panther_gpiod/include/panther_gpiod/gpio_driver.hpp

```diff
@@ -0,0 +1,263 @@
+   */
+  void gpio_monitor_on();
+
+  /**
+   * @brief Disables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_off();
+
+private:
+  std::unique_ptr<gpiod::line_request> create_line_request(gpiod::chip & chip);
+  gpiod::line_settings generate_line_settings(const GPIOInfo & pin_info);
+  GPIOPin get_pin_from_offset(const gpiod::line::offset & offset) const;
+  GPIOInfo & get_gpio_info_ref(const GPIOPin pin);
+  void configure_line_request(
+    gpiod::chip & chip, gpiod::request_builder & builder, GPIOInfo & gpio_info);
+  void monitor_async_events();
+  void configure_rt();
+  void handle_edge_event(const gpiod::edge_event & event);
+  bool is_gpio_monitor_thread_running() const;
+
+  /**
+   * @brief Callback function for GPIO edge events.
+   *
+   * @param gpio_info Information related to the state of the GPIO pin for which the event took
+   * place.
+   */
+  std::function<void(const GPIOInfo & gpio_info)> gpio_edge_event_callback;
+
+  /**
+   * @brief Mapping of GPIO pins to their respective names.
+   *
+   * This map contains the names associated with different GPIO pins.
+   */
+  const std::map<GPIOPin, std::string> pin_names_{
+    {GPIOPin::WATCHDOG, "WATCHDOG"},
+    {GPIOPin::AUX_PW_EN, "AUX_PW_EN"},
+    {GPIOPin::CHRG_DISABLE, "CHRG_DISABLE"},
+    {GPIOPin::CHRG_SENSE, "CHRG_SENSE"},
+    {GPIOPin::DRIVER_EN, "DRIVER_EN"},
+    {GPIOPin::E_STOP_RESET, "E_STOP_RESET"},
+    {GPIOPin::FAN_SW, "FAN_SW"},
+    {GPIOPin::GPOUT1, "GPOUT1"},
+    {GPIOPin::GPOUT2, "GPOUT2"},
+    {GPIOPin::GPIN1, "GPIN1"},
+    {GPIOPin::GPIN2, "GPIN2"},
+    {GPIOPin::LED_SBC_SEL, "LED_SBC_SEL"},
+    {GPIOPin::SHDN_INIT, "SHDN_INIT"},
+    {GPIOPin::VDIG_OFF, "VDIG_OFF"},
+    {GPIOPin::VMOT_ON, "VMOT_ON"},
+  };
+
+  /**
+   * @brief Vector containing GPIO pin configuration information.
+   *
+   * This vector stores information related to GPIO pins such as pin configuration,
+   * direction, value, etc.
+   */
+  std::vector<GPIOInfo> gpio_info_storage_;
+
+  /**
+   * @brief Mutex for managing access to GPIO pin information.
+   *
+   * This shared mutex allows controlling access to GPIO pin information to prevent race conditions.
+   */
+  mutable std::shared_mutex gpio_info_storage_mutex_;
+
+  /**
+   * @brief Request object for controlling GPIO lines.
+   *
+   * This unique pointer manages the request for controlling GPIO lines.
+   */
+  std::unique_ptr<gpiod::line_request> line_request_;
+
+  /**
+   * @brief Thread object for monitoring GPIO events.
+   *
+   * This unique pointer manages the thread responsible for monitoring GPIO events asynchronously.
+   */
+  std::unique_ptr<std::thread> gpio_monitor_thread_;
+  int gpio_monit_thread_sched_priority_;
+  std::atomic<bool> gpio_monitor_thread_enabled_{false};
+  static constexpr unsigned gpio_debounce_period_ = 10;
+  static constexpr unsigned edge_event_buffer_size_ = 2;
+  const std::filesystem::path gpio_chip_path_ = "/dev/gpiochip0";
```

Comment:
shouldn't it be a parameter?

Reply:
I don't have a very strong opinion, but in the Panther Rpi OS image, only this chip seems to be enabled to control the pins described in the GPIOPin enum class. What are your thoughts on this observation?

Reply:
okay, so if there is only one, and we won't change the name, it probably can be hardcoded